### PR TITLE
sd-id128: introduce id128_get_machine() and _at()

### PR DESCRIPTION
**Pull request by yuwata**
split-out of #27081.

### CODE REVIEW

File: src/partition/repart.c [hunk 1]
```diff
@@ -4909,24 +4909,14 @@ static int context_read_seed(Context *context, const char *root) {
...
-                        if (r < 0) {
-                                if (!ERRNO_IS_MACHINE_ID_UNSET(r))
-                                        return log_error_errno(r, "Failed to parse machine ID of image: %m");
+                r = id128_get_machine(root, &context->seed);
+                if (r >= 0)
+                        return 0;
 
-                                log_info("No machine ID set, using randomized partition UUIDs.");
-                        }
+                if (!ERRNO_IS_MACHINE_ID_UNSET(r))
```
Comment (daandemeyer) [inline src/partition/repart.c:4916]: Missing `r < 0` check?

**Review by DaanDeMeyer — APPROVED** [review]
LGTM, one question

File: src/partition/repart.c (hunk 1 above)
Comment (yuwata) [reply src/partition/repart.c:4916]: We have checked `if (r >= 0)` in the above.

**Comment by poettering** [comment]
lgtm

File: src/basic/fd-util.h [hunk 2]
```diff
@@ -101,6 +102,9 @@ int read_nr_open(void);
 int fd_get_diskseq(int fd, uint64_t *ret);
 
 int dir_fd_is_root(int dir_fd);
+static inline int dir_fd_is_root_or_cwd(int dir_fd) {
+        return dir_fd == AT_FDCWD ? true : dir_fd_is_root(dir_fd);
```
Comment (YHNdnzj) [inline src/basic/fd-util.h:106]: Ternary operator is not really needed here?

Comment (yuwata) [reply src/basic/fd-util.h:106]: > Ternary operator is not really needed here?

Hm? What do you mean? Note, dropping `true` triggers warning:
```
../src/basic/fd-util.h: In function ‘dir_fd_is_root_or_cwd’:
../src/basic/fd-util.h:106:36: warning: the omitted middle operand in ‘?:’ will always be ‘true’, suggest explicit middle operand [-Wparentheses]
  106 |         return dir_fd == AT_FDCWD ?: dir_fd_is_root(dir_fd);
      |                                    ^
```

Comment (YHNdnzj) [reply src/basic/fd-util.h:106]: `dir_fd == AT_FDCWD || dir_fd_is_root(dir_fd)` maybe?

Comment (YHNdnzj) [reply src/basic/fd-util.h:106]: Oh, never mind. `dir_fd_is_root` can return negative errno...

Sorry for the noise.

Comment (yuwata) [reply src/basic/fd-util.h:106]: No problem. Thank you for your review!

**Pull request merged**